### ISSUE TITLE
Testing against Mantid

### ISSUE THREAD

**Issue by SimonHeybrock** [scipp/scipp#455]
For neutron-specific code, but also for generic operations, we need to ensure correctness and equivalent behavior to Mantid's implementation of equivalent operations. There may be the occasional difference, but in those cases we must understand why and ensure that this is a deliberate choice.

One way to do such testing is using the `conda` packages for Mantid and scipp, then write tests as follows:

```
workspace -> scipp.compat.mantid.to_dataset -> dataset
   |                                             |
   V                                             V
mantid-alg                                   scipp-alg
   |                                             |
   V                                             V
workspace <- scipp.compat.mantid.from_dataset <- dataset
```

(We should probably avoid using `to_dataset` on the output workspace to ensure that we do not compare a bad conversion result).

**Comment by SimonHeybrock** [comment]
> (We should probably avoid using to_dataset on the output workspace to ensure that we do not compare a bad conversion result).

Not sure I agree with myself here: Converting back to Mantid means:
- We get failures if the converters do not handle all workspace content (which they do not, and cannot or should not).
- We would have to rely on Mantid's `CompareWorkspaces`, which has its quirks (e.g., the many options that can configure its behavior) and is a black box unless we study its implementation details.

**Comment by SimonHeybrock** [comment]
I have created https://github.com/scipp/scipp-ci-mantid for this. `Dockerfile` and scripts for testing should go there.

**Comment by SimonHeybrock** [comment]
@igudich Regarding setting up the tests script, I have something like this in mind:

When implementing a test, I want to implement a class (maybe inheriting a base class?) with a couple of methods:

```python
class MyNiceTest:

    def filenames(self):
        # Return list of filenames required for test
        return ['PG3_4844_event']

    def run_mantid(self, workspace):
        import mantid.simpleapi as mantid
        return mantid.CloneWorkspace(workspace, StoreInADS=False)

    def run_scipp(self, data_array):
        import scipp as sc
        return data_array.copy()
```

The testing framework to set up would then take care of everything else:

- download file
- load file
- run Mantid test
- convert file to scipp
- run scipp test
- convert Mantid result to scipp
- compare

For more complex cases we need a mechanism to provide a custom compare function

**Comment by OwenArnold** [comment]
I agree that the structure of the test file described here https://github.com/scipp/scipp/issues/455#issuecomment-566422754 should make it nice and easy to add.

Rather than doing this with docker. I suggest that we simply do it through conda. Details:

* We create add a new subdirectory containing the tests and infrastructure. I think probably still worth keeping part of the scipp repo?
* We provide an environment file to provide the latest dev `scipp `version as well as the currently used `mantid-framework` package (and any other dependencies needed for test infrastructure)
* We create a new **Master** azure job stage which sets up the environment and runs all tests within using pytest. These tests are likely to be too slow running (download step etc) to run with every PR push.
* We could optionally write a xml report output

**Comment by OwenArnold** [comment]
Update - Testing upon Release, Testing as part of Weekly pipeline.

**Comment by OwenArnold** [comment]
Basic framework and some examples completed as part of https://github.com/scipp/scippneutron/pull/16
